sphinx/util/parallel.py

```
@@ -0,0 +1,106 @@
+            proc.start()
+            result = precv.recv()
+            self.result_queue.put((chunk,) + result)
+            proc.join()
+            semaphore.release()
+
+        # allow only "nproc" worker processes at once
+        semaphore = threading.Semaphore(self.nproc)
+
+        for chunk in self._chunks:
+            yield chunk
+            semaphore.acquire()
```

Pylint (3.3.7)

[refactor] Consider using 'with' for resource-allocating operations

(R1732)